Sources/Session.php

```diff
@@ -95,151 +98,157 @@ function loadSession()
+		// If an insert fails due to a dupe, replace the existing session...
+		$session_update = $smcFunc['db_insert']('replace',
+			'{db_prefix}sessions',
+			array('session_id' => 'string', 'data' => 'string', 'last_update' => 'int'),
+			array($id, $data, time()),
+			array('session_id')
+		);
 
-		$db_connection = smf_db_initiate($db_server, $db_name, $db_user, $db_passwd, $db_prefix, $options);
+		return ($smcFunc['db_affected_rows']() == 0 ? false : true);
 	}
 
-	// If an insert fails due to a dupe, replace the existing session...
-	$session_update = $smcFunc['db_insert']('replace',
-		'{db_prefix}sessions',
-		array('session_id' => 'string', 'data' => 'string', 'last_update' => 'int'),
-		array($session_id, $data, time()),
-		array('session_id')
-	);
-
-	return ($smcFunc['db_affected_rows']() == 0 ? false : true);
-}
+	/**
+	 * Implementation of SessionHandler::destroy() replacing the standard destroy handler.
+	 *
+	 * @param string $session_id The session ID
+	 * @return boolean Whether the session was successfully destroyed
+	 */
+	public function destroy(/*PHP 8.0 string*/$id): bool
+	{
+		global $smcFunc;
 
-/**
- * Implementation of sessionDestroy() replacing the standard destroy handler.
- *
- * @param string $session_id The session ID
- * @return boolean Whether the session was successfully destroyed
- */
-function sessionDestroy($session_id)
-{
-	global $smcFunc;
+		if (!$this->IsValidSessionID($id))
+			return false;
 
-	if (preg_match('~^[A-Za-z0-9,-]{16,64}$~', $session_id) == 0)
-		return false;
+		// Just delete the row...
+		$smcFunc['db_query']('', '
+			DELETE FROM {db_prefix}sessions
+			WHERE session_id = {string:session_id}',
+			array(
+				'session_id' => $id,
+			)
+		);
 
-	// Just delete the row...
-	$smcFunc['db_query']('', '
-		DELETE FROM {db_prefix}sessions
-		WHERE session_id = {string:session_id}',
-		array(
-			'session_id' => $session_id,
-		)
-	);
+		return true;
+	}
 
-	return true;
-}
+	/**
+	 * Implementation of SessionHandler::GC() replacing the standard gc handler.
+	 * Callback for garbage collection.
+	 *
+	 * @param int $max_lifetime The maximum lifetime (in seconds) - prevents deleting of sessions older than this
+	 * @return boolean Whether the option was successful
+	 */
+	#[\ReturnTypeWillChange]
+	public function gc(/*PHP 8.0 int*/$max_lifetime)/*PHP 8.1 : int|false*/
+	{
+		global $modSettings, $smcFunc;
+
+		// Just set to the default or lower?  Ignore it for a higher value. (hopefully)
+		if (!empty($modSettings['databaseSession_lifetime']) && ($max_lifetime <= 1440 || $modSettings['databaseSession_lifetime'] > $max_lifetime))
+			$max_lifetime = max($modSettings['databaseSession_lifetime'], 60);
+
+		// Clean up after yerself ;).
+		$session_update = $smcFunc['db_query']('', '
+			DELETE FROM {db_prefix}sessions
+			WHERE last_update < {int:last_update}',
+			array(
+				'last_update' => time() - $max_lifetime,
+			)
+		);
+
+		return ($smcFunc['db_affected_rows']() == 0 ? false : true);
```

Comment:
`return $smcFunc['db_affected_rows']() !== 0;` or `return $smcFunc['db_affected_rows']();` is shorter

Reply:
I most likely copied from a previous function and didn't check the return type properly.  Fixed